src/core/load-fragment.c

```diff
@@ -4765,6 +4765,37 @@ int config_parse_exec_directories(
         }
 }
 
+int config_parse_exec_quota(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        uint32_t *limit = data;
+        int r;
+
+        if (isempty(rvalue)) {
+                *limit = 0;
```

Comment:
0% is a pretty valid percentage. after all people might want to set up state dirs for which quota refuses any further writing, but the existing files can be read...

maybe the quota should be encoded in a triplet of fields:

```c
uint32_t quota_scale;
uint64_t quota_absolute;
bool quota_set;
```

with the three cases:

1. quota_set is false → no quota configured
2. quota_set is true, and quota_absolute == UINT64_MAX → relative quota is configured in quota_scale
3. quota_set is true, and quota_absolute != UINT64_MAX → absolute quota is configured in quota_absolute

Reply:
this triplet could be it's on struct even:

```c
typedef struct QuotaLimit {
        uint64_t quota_absolute; /* absolute quota in bytes; if UINT64_MAX relative quota configured, see below */
        uint32_t quota_scale;    /* relative quota to backend size, scaled to 0…UINT32_MAX */
        bool quota_set;
} QuotaLimit;
```

and then embed that in ExecDirectory.

(this might become useful elsewhere too).